From 02a760b7c561015996d5068694e99049b6a4c59e Mon Sep 17 00:00:00 2001
From: Philippe Renon <philippe_renon@yahoo.fr>
Date: Thu, 29 Mar 2018 19:15:24 +0200
Subject: [PATCH] msys2: fix linking errors

* add python lib manually
* fix libtool error: add no-undefined flag
* add glib libs
* add gmodule library manually
---
 gi/overrides/Makefile.am | 4 ++--
 plugin/Makefile.am       | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/gi/overrides/Makefile.am b/gi/overrides/Makefile.am
index 64ff170..b431589 100644
--- a/gi/overrides/Makefile.am
+++ b/gi/overrides/Makefile.am
@@ -1,6 +1,6 @@
 common_cflags = $(PYGOBJECT_CFLAGS) $(GST_CFLAGS) -fno-strict-aliasing
-common_libadd = $(GST_LIBS) $(PYGOBJECT_LIBS)
-common_ldflags = -module -avoid-version -shrext $(PYTHON_SO)
+common_libadd = $(PYTHON_LIBS) $(GST_LIBS) $(PYGOBJECT_LIBS)
+common_ldflags = -module -avoid-version -no-undefined -shrext $(PYTHON_SO)
 
 # We install everything in the gi/overrides folder
 pygioverridesdir = $(PYGI_OVERRIDES_DIR)
diff --git a/plugin/Makefile.am b/plugin/Makefile.am
index e6452e1..be8ef8c 100644
--- a/plugin/Makefile.am
+++ b/plugin/Makefile.am
@@ -8,8 +8,8 @@ INCLUDES = $(PYGOBJECT_CFLAGS) $(GST_CFLAGS)\
 	$(PYTHON_INCLUDES)
 
 libgstpython_la_SOURCES = gstpythonplugin.c
-libgstpython_la_LDFLAGS = -avoid-version -shrext $(PYTHON_SO)
-libgstpython_la_LIBADD =  $(PYTHON_LIBS) $(PYGOBJECT_LIBS) $(GST_LIBS)
+libgstpython_la_LDFLAGS = -avoid-version -no-undefined -shrext $(PYTHON_SO)
+libgstpython_la_LIBADD =  $(PYTHON_LIBS) $(PYGOBJECT_LIBS) $(GST_LIBS) -lgmodule-2.0
 libgstpython_la_CFLAGS = $(GST_CFLAGS) $(PYGOBJECT_CFLAGS) $(GST_PLUGINS_BASE_CFLAGS) $(PYTHON_INCLUDES)
 
 EXTRA_DIST = \
-- 
2.16.2

